### PR TITLE
cucumber-expressions: replace special characters

### LINKED ISSUE #103: cucumber-expressions: escape regex special characters
In the javascript variant, the step text `Lucy is at [0, 0]` generates the cucumber expression `Lucy is at [{arg1:int}, {arg2:int}]`. However, this generated cucumber expression does not match the step.

I believe in order to fix this, I believe we need to escape the special characters (in this case the brackets) in the string when creating the regular expression from the string.

See cucumber/cucumber-js#732

### PR DESCRIPTION
**Pull request by charlierudolph**
resolves #103 

Only the JS for now. After initial review, I'll update the java / ruby versions

### CODE REVIEW

File: cucumber-expressions/javascript/test/cucumber_expression_test.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ describe(CucumberExpression.name, () => {
   })
 
   describe('RegExp special characters', () => {
-    ['[', '\\', '^', '$', '?', '*', '+'].forEach((character) => {
+    ['\\', '[', ']', '^', '$', '.', '|', '?', '*', '+'].forEach((character) => {
```
Comment (charlierudolph) [inline cucumber-expressions/javascript/test/cucumber_expression_test.js]: I don't think this change is needed. Moving the backslash to first is fine, but the addition of the closing square bracket, dot, and pipe are unneeded. 

The dot and pipe are tested separately as these tests weren't failing before implement escaping. 

The closing square bracket doesn't need to be tested as that too probably isn't failing and looking up regular expression special characters, it didn't appear that you need to escape the closing square bracket.

**Comment by charlierudolph** [comment]
Also. @aslakhellesoy do you think this should be merged as is or should I update the ruby / java versions in this PR as well?

**Comment by aslakhellesoy** [comment]
We should update the ruby/java versions too, and ideally the shared tests (based on a manually copied text file for now)

File: cucumber-expressions/javascript/test/cucumber_expression_test.js (hunk 1 above)
Comment (aslakhellesoy) [inline cucumber-expressions/javascript/test/cucumber_expression_test.js]: You are right those extra tests are not needed. At first I was surprised some of the characters weren't part of the reused test. Then I was surprised to see some of the characters didn't require escaping. I think adding theses tests may reduce some of the surprises/puzzles in the future - consider it documentation.

Comment (charlierudolph) [inline cucumber-expressions/javascript/test/cucumber_expression_test.js]: I'd rather comment why those tests were excluded then have them in if they aren't needed.

Comment (aslakhellesoy) [inline cucumber-expressions/javascript/test/cucumber_expression_test.js]: Ok, fine with me.

**Pull request merged**